lsp-types/src/Language/LSP/Types/Message.hs

```diff
@@ -371,7 +371,6 @@ data ResponseError =
 
 deriveJSON lspOptions ''ResponseError
 
--- | Either result or error must be Just.
```

Comment:
This is misleading since the former 2 Maybe fields were replaced with Either in 
https://github.com/haskell/lsp/commit/95f2e76d1e8a35cfc6bfdf73eb511768d6cf9392#diff-6031041f17e1defe329987e180cff35227ab23a55bf8fa9415d7ebb4ab5c93d2R409

The comment seems to have returned through merge mistake in 
https://github.com/haskell/lsp/commit/9e78cce4000dde4666cb04fa1fb59a38cc3d8a2a#diff-6031041f17e1defe329987e180cff35227ab23a55bf8fa9415d7ebb4ab5c93d2R368